clients/apple/Shared/Stateful Logic/BillingService.swift

```diff
@@ -0,0 +1,197 @@
+import SwiftLockbookCore
+import SwiftUI
+import StoreKit
+
+public enum StoreError: Error {
+    case failedVerification
+    case noProduct
+}
+
+public enum PurchaseResult {
+    case success
+    case pending
+    case failure
+}
+
+public enum CancelSubscriptionResult {
+    case success
+    case appstoreActionRequired
+}
+
+
+let MONTHLY_SUBSCRIPTION_PRODUCT_ID = "basic.premium"
+
+class BillingService: ObservableObject {
+    let core: LockbookApi
+    
+    var updateListenerTask: Task<Void, Error>? = nil
+
+    var maybeMonthlySubscription: Product? = nil
+    var purchaseResult: PurchaseResult? = nil
+    @Published var cancelSubscriptionResult: CancelSubscriptionResult? = nil
+    
+    var makingPurchaseAttempt = false
+    
+    
+    init(_ core: LockbookApi) {
+        self.core = core
+        updateListenerTask = listenForTransactions()
+
+        Task {
+            await requestProducts()
+        }
```

Comment:
I wonder if we don't want to do this every time the app opens

Reply:
It can be moved, but in the storekit example apple provides, it is called similarly. It shouldn't be a problem to move it though.